app/main.py

```diff
@@ -1,3 +1,56 @@
+import math
+
+
 class Vector:
-    # write your code here
-    pass
+    def __init__(self, x, y):
+        self.x = round(x, 2)
+        self.y = round(y, 2)
+
+    def __add__(self, other):
+        return Vector(self.x + other.x, self.y + other.y)
+
+    def __sub__(self, other):
+        return Vector(self.x - other.x, self.y - other.y)
+
+    def __mul__(self, other):
+        if isinstance(other, (int, float)):
+            return Vector(self.x * other, self.y * other)
+        else:
+            return self.x * other.x + self.y * other.y
```

Comment:
Try to make it without double return

Reply:
I received similar comments from the mentor several times. That's why I make them for you. But it's up to you.

Reply:
sometimes maybe yes, but in this example we will get long expression and we will lose readability of th code.